railgun/controllers/corev1/service_controller.go

```diff
@@ -0,0 +1,105 @@
+package corev1
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/deckgen"
+	"github.com/kong/kubernetes-ingress-controller/pkg/parser"
+	"github.com/kong/kubernetes-ingress-controller/pkg/sendconfig"
+	"github.com/kong/kubernetes-ingress-controller/pkg/store"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/ctrlutils"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/mgrutils"
+)
+
+// -----------------------------------------------------------------------------
+// CoreV1 Service
+// -----------------------------------------------------------------------------
+
+// CoreV1Service reconciles a Service object
+type CoreV1ServiceReconciler struct {
+	client.Client
+
+	Log        logr.Logger
+	Scheme     *runtime.Scheme
+	KongConfig sendconfig.Kong
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *CoreV1ServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// TODO: this is too broad, we need to sweep for Services referred to by other objects we support not all
```

Comment:
note: same as above, unless someone has a strong reason otherwise, I'm intending to do this TODO as a follow up and a KIC 2.0 alpha release blocker (if we go with this direction).

Reply:
8e85bee1f272449dcd0f4d9b37a0b23f6b8be683 resolves this by adding a follow up issue